libs/community/langchain_community/callbacks/upstash_ratelimit_callback.py

```diff
@@ -0,0 +1,211 @@
+"""Ratelimiting Handler to limit requests or tokens"""
+
+import logging
+from typing import Any, Dict, List, Literal, Optional
+
+from langchain_core.callbacks import BaseCallbackHandler
+from langchain_core.outputs import LLMResult
+
+logger = logging.getLogger(__name__)
+try:
+    from upstash_ratelimit import Ratelimit
+except ImportError as error:
+    logger.error(
```

Comment:
Optional imports should have no side-effects for folks that are just importing the file

Side-effects only OK at run time if someone is using this callback (not at import time)